File: trip-kit-booking/src/main/java/com/skedgo/tripkit/booking/mybookings/MyBookingsConfirmationResponse.java
package com.skedgo.tripkit.booking.mybookings;

import com.google.gson.annotations.JsonAdapter;
import com.skedgo.tripkit.common.model.BookingConfirmation;

import org.immutables.value.Value;

import java.util.List;

import androidx.annotation.Nullable;

import static org.immutables.gson.Gson.TypeAdapters;
import static org.immutables.value.Value.Immutable;
import static org.immutables.value.Value.Style;

@Immutable
@TypeAdapters
@Style(passAnnotations = JsonAdapter.class)
@JsonAdapter(GsonAdaptersMyBookingsConfirmationResponse.class)
public abstract class MyBookingsConfirmationResponse {

    public abstract BookingConfirmation confirmation();

    @Value.Default
    public int index() {
        return 0;
    }

    @Value.Default
    public int time() {
        return 0;
    }

    public abstract @Nullable List<String> trips();

    public abstract @Nullable String mode();

}

File: trip-kit-booking/src/main/java/com/skedgo/tripkit/booking/mybookings/MyBookingsConfirmationResponse.kt
package com.skedgo.tripkit.booking.mybookings

import com.google.gson.annotations.JsonAdapter
import com.skedgo.tripkit.common.model.booking.confirmation.BookingConfirmation
import org.immutables.gson.Gson.TypeAdapters
import org.immutables.value.Value.Default
import org.immutables.value.Value.Immutable
import org.immutables.value.Value.Style

@Immutable
@TypeAdapters
@Style(passAnnotations = [JsonAdapter::class])
@JsonAdapter(
    GsonAdaptersMyBookingsConfirmationResponse::class
)
abstract class MyBookingsConfirmationResponse {
    abstract fun confirmation(): BookingConfirmation

    @Default
    open fun index(): Int {
        return 0
    }

    @Default
    open fun time(): Int {
        return 0
    }

    abstract fun trips(): List<String?>?

    abstract fun mode(): String?
}
